pkg/tls/ca.go

```diff
@@ -167,6 +175,30 @@ func (ca *CA) createTemplate(publicKey *ecdsa.PublicKey) x509.Certificate {
 	}
 }
 
+// PEMEncodeCert returns the PEM encoding of cert.
+func PEMEncodeCert(cert []byte) ([]byte, error) {
+	buf := &bytes.Buffer{}
+	err := pem.Encode(buf, &pem.Block{Type: "CERTIFICATE", Bytes: cert})
+	return buf.Bytes(), err
+}
+
+// PEMEncodeKey returns the PEM encoding of key.
+func PEMEncodeKey(key []byte, keyType string) ([]byte, error) {
```

Comment:
Same comment as above for this encoded private key func. Also, it looks like we were only ever calling this func with `KeyTypeECDSA`. Since we're not using it, let's rip out support for different key types:

```
func (cpk *CertificateAndPrivateKey) EncodedPrivateKey() ([]byte, error) {
	buf := &bytes.Buffer{}
	err := pem.Encode(buf, &pem.Block{Type: "EC PRIVATE KEY", Bytes: cpk.PrivateKey})
	return buf.Bytes(), err
}
```

That will also allow you to remove the `KeyType*` consts at the top of this file.

Reply:
:cool: